pilot/api/data.py

```diff
@@ -484,12 +595,26 @@ def resolve_surl(self, fspec, protocol, ddmconf, **kwargs):
             raise PilotException('Failed to resolve ddmendpoint by name=%s' % fspec.ddmendpoint)
 
         if not ddm.is_deterministic:
-            raise PilotException('resolve_surl(): Failed to construct SURL for non deterministic ddm=%s: NOT IMPLEMENTED', fspec.ddmendpoint)
-
-        surl = protocol.get('endpoint', '') + os.path.join(protocol.get('path', ''), self.get_path(fspec.scope, fspec.lfn))
+            if ddm.type in ['OS_ES', 'OS_LOGS']:
+                surl = self.resolve_surl_os(fspec, protocol, ddm)
+            else:
+                raise PilotException('resolve_surl(): Failed to construct SURL for non deterministic ddm=%s: NOT IMPLEMENTED', fspec.ddmendpoint)
+        else:
+            surl = protocol.get('endpoint', '') + os.path.join(protocol.get('path', ''), self.get_path(fspec.scope, fspec.lfn))
 
         return {'surl': surl}
 
+    def fill_storage_id(self, files):
```

Comment:
Aha, here you fill storage_id and then later will try to dance in dark to apply more post fix actions, why?

why do we need the `.storage_id` in general once we have already `.ddmendpoint`